src/lib/y2storage/setup_error.rb

```diff
@@ -0,0 +1,228 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast/i18n"
+require "y2storage"
+
+module Y2Storage
+  # Class to represent storage setup error
+  class SetupError
+    include Yast::I18n
+
+    # @return [VolumeSpecification]
+    attr_reader :missing_volume
+
+    # Constructor
+    #
+    # @param message [String, nil] error message
+    # @param missing_volume [VolumeSpecification, nil] missing volume that causes the error
+    def initialize(message: nil, missing_volume: nil)
+      textdomain "storage"
+      @message = message
+      @missing_volume = missing_volume
+    end
+
+    # Error message
+    #
+    # @note If no message was indicated, it can be generated from the missing volume.
+    #
+    # @return [String, nil]
+    def message
+      message = @message
+      message ||= message_for_missing_volume if missing_volume
+      message
+    end
+
+  private
+
+    # Error text for the missing volume
+    #
+    # @return [String]
+    def message_for_missing_volume
+      if mount_point_info
+        message_with_mount_point
+      else
+        message_without_mount_point
+      end
+    end
+
+    # Error text when the missing volume has mount point
+    #
+    # @return [String]
+    def message_with_mount_point
+      if partition_id_info && fs_types_info
+        message_with_mount_point_and_partition_id_and_fs
+      elsif partition_id_info
+        message_with_mount_point_and_partition_id
+      elsif fs_types_info
+        message_with_mount_point_and_fs
+      else
+        message_with_mount_point_default
+      end
+    end
+
+    # Error text when the missing volume does not have mount point
+    #
+    # @return [String]
+    def message_without_mount_point
+      if partition_id_info && fs_types_info
+        message_with_partition_id_and_fs
+      elsif partition_id_info
+        message_with_partition_id
+      elsif fs_types_info
+        message_with_fs
+      else
+        message_without_mount_point_default
+      end
+    end
+
+    # @return [String]
+    def message_with_mount_point_and_partition_id_and_fs
+      # TRANSLATORS: error message, where %{mount_point} is repalaced by a mount point
```

Comment:
typo: repalaced... unless it means we build a new palace for it.

Reply:
:-)